resources/services/ec2/vpc_endpoint_service_configuration.go

```diff
@@ -0,0 +1,176 @@
+				Description: "The endpoint service verification type, for example TXT.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("PrivateDnsNameConfiguration.Type"),
+			},
+			{
+				Name:        "private_dns_name_configuration_value",
+				Description: "The value the service provider adds to the private DNS name domain record before verification.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("PrivateDnsNameConfiguration.Value"),
+			},
+			{
+				Name:        "service_id",
+				Description: "The ID of the service.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "service_name",
+				Description: "The name of the service.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "service_state",
+				Description: "The service state.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "service_type",
+				Description: "The type of service.",
+				Type:        schema.TypeStringArray,
+				Resolver:    resolveEc2VpcEndpointServiceConfigurationServiceType,
+			},
+			{
+				Name:        "tags",
+				Description: "Any tags assigned to the service.",
+				Type:        schema.TypeJSON,
+				Resolver:    resolveEc2VpcEndpointServiceConfigurationTags,
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchEc2VpcEndpointServiceConfigurations(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
+	var config ec2.DescribeVpcEndpointServiceConfigurationsInput
+	c := meta.(*client.Client)
+	svc := c.Services().EC2
+	for {
+		output, err := svc.DescribeVpcEndpointServiceConfigurations(ctx, &config, func(options *ec2.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		res <- output.ServiceConfigurations
+		if aws.ToString(output.NextToken) == "" {
+			break
+		}
+		config.NextToken = output.NextToken
+	}
+	return nil
+}
+func resolveEc2VpcEndpointServiceConfigurationServiceType(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ServiceConfiguration)
+	st := make([]string, 0, len(r.ServiceType))
+	for _, std := range r.ServiceType {
+		st = append(st, string(std.ServiceType))
+	}
+	return diag.WrapError(resource.Set("service_type", st))
+}
+func resolveEc2VpcEndpointServiceConfigurationTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ServiceConfiguration)
+	tags := map[string]*string{}
+	for _, t := range r.Tags {
+		tags[*t.Key] = t.Value
+	}
+	return diag.WrapError(resource.Set("tags", tags))
+}
```

Comment:
Client function can help with this:
```suggestion
func resolveEc2VpcEndpointServiceConfigurationTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
	r := resource.Item.(types.ServiceConfiguration)
	return diag.WrapError(resource.Set(c.Name, client.TagsToMap(r.Tags)))
}
```

Reply:
Ah! Didn't know that was there, just added that 👍